crates/mir/src/lower/function.rs

```diff
@@ -668,6 +668,25 @@ impl<'db, 'a> BodyLowerHelper<'db, 'a> {
                 self.builder.aggregate_construct(ty, args, expr.into())
             }
 
+            ast::Expr::Repeat { value, len: _ } => {
+                let array_type = if let Type::Array(array_type) = self.analyzer_body.expressions
+                    [&expr.id]
+                    .typ
+                    .typ(self.db.upcast())
+                {
+                    array_type
+                } else {
+                    panic!("not an array");
+                };
+
+                let args = (0..array_type.size)
+                    .into_iter()
+                    .map(|_| self.lower_expr_to_value(value))
+                    .collect();
```

Comment:
Should an expression for array elements value be evaluated N times? At least, Rust evaluates it just once. I think Rust can do this because a type in the array repetition initialization must implement `Copy`. Also, we could assume that all types in `Fe` implements `Copy`. So it would be fine to evaluate it just once, and it would greatly reduce gas usage when the expression is expensive.
But from a language-design perspective, I'm not sure which is easier for users to understand the semantics. It might be difficult for those who are addicted to list comprehension. On the other hand, I wonder a list comprehension might be confusing when the language allows side effects.

@sbillig, @cburgdorf  I pinged you two because this discussion would affect the language specification.

Reply:
I'll change it to evaluate once